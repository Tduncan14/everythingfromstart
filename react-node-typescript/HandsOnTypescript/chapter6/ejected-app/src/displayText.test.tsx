import React from 'react';
import { render,fireEvent } from '@testing-library/react';
import DisplayText from './displayText';
import"@testing-library/jest-dom/extend-expect";



describe("Test DisplayText",() => {

    it("renders without crashing",() => {

        const {baseElement} = render(<DisplayText />);
        console.log(baseElement.innerHTML);
        expect(baseElement).toBeInTheDocument();

    });

    it("recieves input text",() => {

        const testuser = "testuser";
        const{getByTestId} = render(<DisplayText />);
        const input = getByTestId("user-input");

        fireEvent.change(input,{target: {value:testuser}});
        expect(input).toBeInTheDocument()
        expect(input).toHaveValue(testuser)
    })

    it("shows welcome message", () => {
                const username = "testuser";
                const msg = `Welcome to React testing,${username}`;
                const { getByTestId } = render(<DisplayText />);
                const input = getByTestId("user-input");
                const label = getByTestId("final-msg");
                fireEvent.change(input, { target: { value:username } });
                const btn = getByTestId("input-submit");
                fireEvent.click(btn);
                
                expect(label).toBeInTheDocument();
                expect(label.innerHTML).toBe(msg);
            });

    

    it("matches snapshot", () => {

        const {baseElement} = render(<DisplayText />);
        expect(baseElement).toMatchSnapshot();


    })


    


})