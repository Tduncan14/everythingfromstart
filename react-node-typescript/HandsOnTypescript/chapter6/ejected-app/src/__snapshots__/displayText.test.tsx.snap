// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DisplayText matches snapshot 1`] = `
<body>
  <div>
    <form>
      <div>
        <label>
          Enter your name
        </label>
      </div>
      <div>
        <input
          data-testid="user-input"
          value=""
        />
      </div>
      <div>
        <button
          data-testid="input-submit"
        >
          show message
        </button>
      </div>
      <div>
        <label
          data-testid="final-msg"
        >
           
        </label>
      </div>
      <div>
        this is just a test entry
      </div>
    </form>
  </div>
</body>
`;
